crates/chain/src/keychain/txout_index.rs

```diff
@@ -158,13 +261,30 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
     ///
     /// This will panic if a different `descriptor` is introduced to the same `keychain`.
     pub fn add_keychain(&mut self, keychain: K, descriptor: Descriptor<DescriptorPublicKey>) {
```

Comment:
> Agreed. Can I open an issue for this? I think it should be easy to tackle for new contributors, is it ok if this is not fixed in this PR?

I'm against this because the main purpose of this PR is to be able to persist descriptors (albeit, with structures that aren't error-prone) and so descriptors SHOULD go in the changesets whenever added.

In terms of accidental replacing, I think we should have two methods; `add_keychain` and `add_or_replace_keychain`. `add_keychain` does not allow for replacing keychains and returns `Error<ChangeSet, {already-existing-keychain}>`.